li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -433,6 +442,15 @@ public long getRecordsSkipped() {
     return recordsSkipped;
   }
 
+  public long getConsumerOffsetWatermarkSpan() {
+    long span = 0;
+    for (TopicPartition tp : knownPartitions()) {
```

Comment:
This is probably also not safe for multithreaded access either since the partition assignment can change while this is running.

Reply:
true, but these are sensor readings, they can mess up. i think some kafka sensors already do. also the value of knownPartitions() may not hold for the entire iteration

Reply:
Yes, and I've had to fix ConcurrentModificationException stack traces many times as they will occasionally fill users' logs.

Reply:
@smccauliff Are you saying the access to `knownPartitions()` is not thread safe? 

@radai-rosenblatt It should be fine if the assigned partitions don't hold for the entire iteration. The value returned by `deliveryMessageOffsetTracker` is 0, if a topic partition is not being tracked.

Reply:
This is what I see:
```java
  public Set<TopicPartition> knownPartitions() {
    return Collections.unmodifiableSet(_partitionConsumerHighWatermark.keySet());
  }
```
If _partitionConsumerHighWatermark is modified during the computation, by the consumer thread, then the metrics reporting thread can get ConcurrentModificationException.  I've seen similar problems before.  Either the code needs to be made thread safe or ConcurrentModificationException needs to be caught and then the computation needs to be retried.

Reply:
I made the change with a memcpy instead of handling `ConcurrentModificationException`. Let me know if that is ok.